### DISCUSSION TITLE
Unable to use 'devs' command after installation.

### DISCUSSION THREAD

**Discussion by Tanmay53** [microsoft/devicescript#438] [Q&A] [answered]
I started the installation as per the instructions in [CONTRIBUTING.md](https://github.com/microsoft/devicescript/blob/main/CONTRIBUTING.md)

I am stuck at the point where it says

-   run `devs run devs/samples/something.ts` - this will execute given DeviceScript program using the WASM binary

I get the following error :
![image](https://github.com/microsoft/devicescript/assets/52573153/1d38435d-bc03-4b03-a373-a0109e3331b0)

System Info:
Window 11
Node 18.15.0

**Comment by pelikhan** [comment, marked as answer]
Ouchy, we haven't tested this enough on windows. Is this powershell? Trying adding `yarn`:

    yarn devs crun devs/samples/something.tsp

**Reply by pelikhan** [reply]
Actually, CONTRIBUTING is quite outdated. Give me 10 minute to refresh it.

**Reply by pelikhan** [reply]
I've done a refresh pass on the instructions.

**Comment by pelikhan** [comment]
If I may ask, which feature are you trying to work on ? It might be that running the vscode extension is more effective for you.

**Reply by Tanmay53** [reply]
Trying to setup the project and test if it's running on my system.

**Comment by gsteiert** [comment]
I get something similar on a brand new Ubuntu install.  I've tried in VS Code and command line.
From command line, npm says the install completed, but there is no devs command available.  I feel like I am missing something obvious.
This is ubuntu 22.04, node v20.4.0

**Reply by pelikhan** [reply]
Are you working from our repo or are you using the extension?

Also did "npm install" of the @devicescript/cli succeed? For some distros, the SerialPort package fails to install because there is no prebuilt image.

**Reply by pelikhan** [reply]
(We don't run our tests against node 20. Might be dragons.)

What does your command failure look like?

**Reply by louisvangeldrop** [reply]
To prevent this problem I install @devicescript/cli always in the global npm repository using: npm install -g @devicescript/cli.

**Reply by mmoskal** [reply]
Note that globally installed `devs` command line [will still use](https://github.com/microsoft/devicescript/blob/main/cli/devicescript#L4) locally installed `@devicescript/cli` if it finds it. It always prints out version number on start for this reason.

**Reply by gsteiert** [reply]
It could not find devs, like a path error. I'll try installing with -g

**Reply by pelikhan** [reply]
you'll get more luck do "yarn devs" as yarn has better heuristics to find the locally installed tools.
